conu/apidefs/filesystem.py

```diff
@@ -0,0 +1,141 @@
+        self._mount_point = mount_point
+        self.mount_point_provided = False
+
+    @property
+    def mount_point(self):
+        if self._mount_point is None:
+            self._mount_point = mkdtemp(prefix="conu")
+            self.mount_point_provided = True
+        return self._mount_point
+
+    def __enter__(self):
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        if self.mount_point_provided:
+            os.rmdir(self.mount_point)
+
+    def p(self, path):
+        """
+        provide absolute path within the container
+
+        :param path: path with container
+        :return: str
+        """
+        if path.startswith("/"):
+            path = path[1:]
+        p = os.path.join(self.mount_point, path)
+        logger.debug("path = %s", p)
+        return p
+
+    def copy_to(self, src, dest):
+        """
+        copy a file or a directory from host system to a container -- don't implement for images,
+        those are immutable
+
+        :param src: str, path to a file or a directory on host system
+        :param dest: str, path to a file or a directory within container
+        :return: None
+        """
+        raise NotImplementedError("copy_to method is not implemented")
+
+    def copy_from(self, src, dest):
+        """
+        copy a file or a directory from container or image to host system
+
+        :param src: str, path to a file or a directory within container or image
+        :param dest: str, path to a file or a directory on host system
+        :return: None
+        """
+        p = self.p(src)
+        if os.path.isfile(src):
+            logger.info("copying file %s to %s", p, dest)
+            shutil.copy2(p, dest)
+        else:
+            logger.info("copying directory %s to %s", p, dest)
+            shutil.copytree(p, dest)
+
+    def read_file(self, file_path):
+        """
+        read file specified via 'file_path' and return its content - raises an ConuException if
+        there is an issue with read the file
+
+        :param file_path: str, path to the file to read
+        :return: str (not bytes), content of the file
+        """
+        with open(self.p(file_path)) as fd:
+            return fd.read()
+
+    def get_file(self, file_path, mode="r"):
+        """
+        provide File object specified via 'file_path'
+
+        :param file_path: str, path to the file
+        :param mode: str, mode used when opening the file
+        :return: File instance
+        """
+        return open(self.p(file_path), mode=mode)
+
+    def file_is_present(self, file_path):
+        """
+        check if file 'file_path' is present, raises IOError if file_path
+        is not a file
+
+        :param file_path: str, path to the file
+        :return: True if file exists, False if file does not exist
+        """
+        p = self.p(file_path)
+        if not os.path.exists(p):
+            return False
+        if not os.path.isfile(p):
```

Comment:
The condition sound reasonable, but personally, I would remove it. I should know what I check, like I will not check directory if I know the file is really file. Function should check if file really exists and nothing else.

Reply:
> I should know what I check, like I will not check directory if I know the file is really file.

I completely disagree. You may think you know what you check, but in reality, there could be a bug and the file may very well turn into a directory. In such case, the function named `file_is_present`, in my opinion, should not return True nor False and should raise and exc since it's not operating on files any more.

If we implemented what you propose, then it's enough to have a single `exists()` function.

@jscotka @bkabrda @dhodovsk opinions?

Reply:
raise sound good and I am find with that. But I would prefer `True` or raise

Reply:
> But I would prefer True or raise

You are not helping here. What do you want me to do?